src/plugins/intel_cpu/tests/unit/shape_inference_test/string_tensor_unpack_shape_inference_test.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/test_assertions.hpp"
+#include "utils.hpp"
+#include "string_tensor_unpack_shape_inference.hpp"
+
+using namespace ov;
+using namespace ov::intel_cpu;
+using ov::op::v0::Constant;
+using ov::op::v0::Parameter;
+
+static size_t get_character_count(const std::vector<std::string>& vec) {
+    size_t count = 0;
+    for (const auto& str : vec) {
+        count += str.size();
+    }
+    return count;
+}
+
+class StringTensorUnpackStaticTestSuite : public ::testing::TestWithParam<std::tuple<
+                                                              std::vector<std::string>,         // input data
+                                                              Shape                             // input shape
+                                                              >> {};
+
+class StringTensorUnpackStaticShapeInferenceTest: public OpStaticShapeInferenceTest<op::v15::StringTensorUnpack> {};
+
+TEST_F(StringTensorUnpackStaticShapeInferenceTest, data_from_tensor_accessor) {
+    const auto data = std::make_shared<Parameter>(element::string, ov::PartialShape::dynamic());
+    const auto op = make_op(data);
+    std::string data_val[] = {"Intel", "OpenVINO"};
+    auto const_inputs = std::unordered_map<size_t, Tensor>{{0, {element::string, Shape{2}, data_val}}};
+
+    const auto input_shapes = ShapeVector{Shape{2}};
```

Comment:
Can this test with tensor accessor be parametrized with different shapes and values, the same way, as the test with Constant input below?

Reply:
Added new `tensor_accessor` tests.